### PR TITLE
[doc] Update internals/bootstrapping instructions.

### PR DESCRIPTION
**Pull request by fnichol**
This change updates the `docs/internals/#bootstrap-internals` page on
the Habitat website with an aim to make it more currently accurate and
a basis for testing more aggressive base plans updates. While still not
100% complete (or possibly accurate), it is more so than what currently
exists.

Signed-off-by: Fletcher Nichol <fnichol@nichol.ca>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by fnichol** [comment]
@thesentinels acceptance

**Comment by thesentinels** [comment]
:metal: I am sending your branch against master to our acceptance environment.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started the deployment of acceptance](https://travis-ci.org/habitat-sh/habitat/builds/359937880).

File: www/source/partials/docs/_internals-bootstrapping.html.md.erb [hunk 1]
```diff
@@ -5,164 +5,339 @@ This document provides developer documentation on how the Habitat system becomes
...
+```
 
 Finally, we place a recent last-known-good copy of the `hab` binary inside `tools/bin`.
 
-The entire tarball of bootstrap "tools" lives inside the [stage1 studio](https://habitat-studio-stage1.s3.amazonaws.com/habitat-studio-stage1-20160612022150.tar.xz) tarball. This should be unpacked into `/tools` on a Linux host that will serve as the build environment until the system is self-sustaining through the rest of this procedure.
+The entire tarball of bootstrap "tools" lives inside the [stage1 studio](http://s3-us-west-2.amazonaws.com/habitat-studio-stage1/habitat-studio-stage1-20180312233639.tar.xz) tarball. This should be unpacked into `/tools` on a Linux host that will serve as the build environment until the system is self-sustaining through the rest of this procedure.
+
+Note: In order to build large amounts of softwar in the fastest way possible, we recommend you use a powerful machine, such as an `c4.4xlarge` on Amazon Web Services (AWS). While it may be possible to run these builds in Docker containers, in workstation virtual machines, or older hosts, it's well worth it to spend a few dollars to save potentially dozens of hours of wait time.
```
Comment (christophermaier) [inline www/source/partials/docs/_internals-bootstrapping.html.md.erb]: "softwar" => "software"

Comment (christophermaier) [reply www/source/partials/docs/_internals-bootstrapping.html.md.erb]: "such as an `c4.4xlarge`" => "such as a `c4.4xlarge`"

Comment (christophermaier) [reply www/source/partials/docs/_internals-bootstrapping.html.md.erb]: "or older hosts" => "or on older hosts"

File: www/source/partials/docs/_internals-bootstrapping.html.md.erb [hunk 2]
```diff
@@ -5,164 +5,339 @@ This document provides developer documentation on how the Habitat system becomes
...
 
 ## Part II: Stage 0
 
+### Creating The stage1 tarball
```
Comment (christophermaier) [inline www/source/partials/docs/_internals-bootstrapping.html.md.erb]: The header says "Stage 0", but the subsequent text talks about "Stage 1"; which is correct?

File: www/source/partials/docs/_internals-bootstrapping.html.md.erb [hunk 3]
```diff
@@ -5,164 +5,339 @@ This document provides developer documentation on how the Habitat system becomes
...
+$ ./core-plans/bin/bootstrap/stage3-build-remaining-plans.sh stage3_world_order.working
 ```
 
-## Part V: Remaining packages in world
+If there are failures as you work through the remaining world, you may discover this is due to a) an older version of the software that can't build against a newer/differnt compiler, b) a failure that is currently in master and not yet caught (ex: source url now returns a 404), c) other issues which require more debugging and work. You can try to fix the package at this moment and re-run the `stage3-build-remaining-plan.sh` which will resume your failed build and continue. Alternatively, you could remove that Plan out of the rebuild set and deal with it separately. The catch here is that your failing package might become a dependency of another package later on, so you would want to remove those as well. Thankfully, there's a way you can get this list, by running another program in the `core-plans` repository. Note that you want to run this in the same common parent directory containing the `core-plans` and `habitat` git clones, and also outside of a Studio. Let's assume that the plan is for `core/doxygen`. You can run:
```
Comment (christophermaier) [inline www/source/partials/docs/_internals-bootstrapping.html.md.erb]: "newer/differnt" => "newer / different"

Comment (christophermaier) [reply www/source/partials/docs/_internals-bootstrapping.html.md.erb]: "not yet caught" => ??

Comment (christophermaier) [reply www/source/partials/docs/_internals-bootstrapping.html.md.erb]: Re: the suggestion to do these builds in a directory that has both the habitat and core-plans repos... it'd be good to call this out earlier, and more prominently.

File: www/source/partials/docs/_internals-bootstrapping.html.md.erb [hunk 4]
```diff
@@ -5,164 +5,339 @@ This document provides developer documentation on how the Habitat system becomes
...
 
+Now that we're in a Studio, it's been set up with an empty artifact cache. We're going to set things up so that our install logic won't need to call out to a Builder API:
+
 ```studio
-$ $BUILD libarchive
-$ $BUILD libsodium
-$ $BUILD zeromq
-$ $BUILD ../components/depot
+# Copy all stage1 packages into the artifact cache
+$ cp ./results-stage1/*.hart /hab/cache/artifacts/
```
Comment (rsertelon) [inline www/source/partials/docs/_internals-bootstrapping.html.md.erb]: The `*` character does not appear on acceptance. Maybe the template engine swallows it? It's not practical for copy/pasting lines :) this happens in several places in code snippets.

Comment (christophermaier) [reply www/source/partials/docs/_internals-bootstrapping.html.md.erb]: Oooh, that's not good 😦 

My guess is that it's due to the templating engine, similar to https://github.com/habitat-sh/habitat/issues/4587.

Wonder if `\*` works here?

Pinging @cnunciato for other ideas.

**Comment by christophermaier** [comment]
@fnichol Is this still relevant?

**Comment by fnichol** [comment]
Yeah it is--if I recall the largest amount of work was ensuring that newline escapes (i.e. `\`) were rendering correctly via middleman. I will do my best to address the updates and at least take a pass at looking at the rendering issues. Worst case is that we file an issue to address the rendering, but these instructions are current and up to date in the source tree. That is, until we add in the updated work that has happened since June 😄

**Pull request merged**